lib/rules/no-legacy-test-waiters.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+const { getImportIdentifier } = require('../utils/import');
+
+const LEGACY_TEST_WAITER_FUNCTIONS = ['registerWaiter', 'unregisterWaiter'];
+const ERROR_MESSAGE =
+  'The use of the legacy test waiters API is not allowed. Please use the new ember-test-waiters (https://github.com/emberjs/ember-test-waiters) APIs instead.';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow the use of the legacy test waiter APIs.',
+      category: 'Testing',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-legacy-test-waiters.md',
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    let testWaitersIdentifiers = [];
+
+    return {
+      ImportDeclaration(node) {
+        if (node.source.value !== '@ember/test') {
+          return;
+        }
+
+        testWaitersIdentifiers = LEGACY_TEST_WAITER_FUNCTIONS.map(fn =>
+          getImportIdentifier(node, '@ember/test', fn)
+        );
+
+        if (testWaitersIdentifiers.length > 0) {
+          context.report(node, ERROR_MESSAGE);
```

Comment:
Is it necessary to report a violation on the import statement? Isn't reporting the function call sufficient? If can be kind of annoying to report two violations for a single usage, unless there's a good reason to do this.

Reply:
Well my thought is to report on the import so people won't forget to clean that up too. You don't want imports dangling around, if you're not using an editor/lint rule that's helping identify that.

Reply:
Pretty much everybody should be using the [no-unused-vars](https://eslint.org/docs/rules/no-unused-vars) eslint recommended rule that will catch this problem. I generally don't think our lint rules should be responsible for that, it's a separate issue. In addition, somebody could be importing something else from `@ember/test` (i.e. `import { stillAllowedFunction } from @ember/test`) that is still allowed, so we shouldn't flag the entire import statement anyway.

Reply:
The last point is fair - and is the most compelling reason for me to remove this :)